### PR TITLE
fetch company data fram api

### LINKED ISSUE #708: Publish new company list in preview mode, only showing the large cap companies unless a special URL query parameter is present
This needs to be fixed as part of #704

### LINKED ISSUE #709: Add support to show combined scope 1 and 2 emissions in company list
**Description**
Add support to show combined scope 1 and 2 emissions in company list so that correct data is shown for companies that report these as a total

**When should it be done?**
Urgent!

### PR DESCRIPTION
**Pull request by elvbom**
fix #709 

## TODO: 
fix #708

### CODE REVIEW

File: utils/companyDataService.tsx [hunk 1]
```diff
@@ -1,43 +1,57 @@
...
+  // Load companies data asynchronously
+  private async loadCompanies() {
+    try {
+      const response = await fetch('https://api.klimatkollen.se/api/companies')
+
+      if (!response.ok) {
+        throw new Error('Failed to fetch data from the API')
+      }
+
+      const jsonData = await response.json()
```
Comment (Greenheart) [inline utils/companyDataService.tsx:20]: This is not thorough review, but it looks good from a quick look! :)

We get an error in the vercel build regarding the response type here. We could use the same response type as the one from https://github.com/Klimatbyran/beta/blob/d27544f1aedaa5128eae80052a85a74bdfff06f2/src/data/companies.ts#L4 to fix this issue and allow the build to complete.

```sh

Type error: 'jsonData' is of type 'unknown'.
  20 |       const jsonData = await response.json()
  21 |
> 22 |       this.companies = jsonData.map((data: any) => {
```

**Comment by elvbom** [comment]
Please double check that the API emission values are represented correctly before merge - looked good to me during development but work was done on the database when I completed this couldn't do a final check.

File: package.json [hunk 2]
```diff
@@ -27,6 +27,7 @@
     "next": "14.2.3",
     "next-i18next": "^15.2.0",
     "nextjs-progressbar": "^0.0.16",
+    "node-fetch": "^3.3.2",
```
[the remark this answers is not in the document]
Comment (Greenheart) [reply package.json:30]: We'll skip this for now

File: utils/companyDataService.tsx [hunk 3]
```diff
@@ -1,43 +1,63 @@
...
+        } as Company
       })
+    } catch (error) {
+      throw new Error('Failed to retrieve company data from the API')
+    }
   }
 
-  public getCompanies(): Array<Company> {
+  public async getCompanies(): Promise<Array<Company>> {
     if (this.companies.length < 1) {
```
[the remark this answers is not in the document]
Comment (Greenheart) [reply utils/companyDataService.tsx:52]: We'll skip this for now

**Review by Greenheart — APPROVED** [review]
Thanks for fixing this! :)

**Comment by elvbom** [comment]
FIx #709

**Pull request merged**